mqtt/mqtt-edgehub/src/auth/authorization/edgehub.rs

```diff
@@ -122,17 +122,30 @@ where
             ),
             // allow authenticated clients with client_id == auth_id and accessing its own IoTHub topic
             AuthId::Identity(identity) if identity == activity.client_id() => {
-                if self.is_iothub_topic_allowed(activity.client_id(), topic)
-                    && self.check_authorized_cache(activity.client_id(), topic)
+                let iothub_topic_allowed = self.is_iothub_topic_allowed(activity.client_id(), topic);
+                let exists_in_authorized_cache = self.is_iothub_topic_allowed(activity.client_id(), topic);
+                if iothub_topic_allowed && exists_in_authorized_cache
                 {
                     Authorization::Allowed
                 } else {
                     // check if iothub policy is overridden by a custom policy.
                     if Authorization::Allowed == self.inner.authorize(activity)? {
                         Authorization::Allowed
                     } else {
+                        let iothub_topic_reason = "IoT Hub topic is not allowed. Client must access its own IoT Hub topics only";
+                        let exists_in_cache_reason = "Identity" + identity.to_string() + " does not exist in authorized cache";
```

Comment:
I guess that could be more customer friendly message?
```suggestion
                        let exists_in_cache_reason = format!("Identity {0} is not authorized to access topic {1}", identity, topic);
```